contracts/modules/LoanOpenings/LoanOpenings.sol

```diff
@@ -79,6 +80,55 @@ contract LoanOpenings is State, LoanOpeningsEvents, VaultController, InterestHan
             loanParams[loanParamsLocal.id] = loanParamsLocal;
         }
     }
+
+    function swapLoanCollateral(
+        bytes32 loanId,
+        address newCollateralToken,
+        bytes calldata loanDataBytes)
+        external
+        nonReentrant
+        pausable
+    {
+        require(supportedTokens[newCollateralToken], "unsupported");
+        Loan storage loanLocal = loans[loanId];
+        require(loanLocal.active, "loan is closed");
+        require(
+            msg.sender == loanLocal.borrower ||
+            delegatedManagers[loanLocal.id][msg.sender],
+            "unauthorized"
+        );
+        LoanParams storage loanParamsLocal = loanParams[loanLocal.loanParamsId];
```

Comment:
why not `memory` and use it as variables?

Reply:
or you seems to need only two variables from the struct. so you sload  only them ?

Reply:
loan params could be memory as well but i just store the values in variables anyways which are in memory.